tests/test_hydra.py

```diff
@@ -1047,6 +1047,29 @@ def test_run_pass_list(self, cmd_base: List[str], tmpdir: Any) -> None:
         ret, _err = get_run_output(cmd)
         assert OmegaConf.create(ret) == OmegaConf.create(expected)
 
+    def test_multirun_dict_keys(self, cmd_base: List[str], tmpdir: Any) -> None:
```

Comment:
This test, and test_overrides_dict_keys - are very high level.
you have some low level tests in test_overrides_parser and you are jumping straight into testing as processes.

If I understand the purpose of those tests correctly (and it's hard because they are so high level), I think should should be replaced by lower level config composition test in test_config_loader.py.

Reply:
So, to clarify the motivation: those two new tests are meant to test the changes in `types.py` (which are not tested by the low level grammar tests)

The second one (`test_overrides_dict_keys`) should probably indeed be a lower level test. I wasn't familiar with the various Hydra tests and couldn't figure out where the basic override functionalities were being tested. I just moved it to `test_config_loader.py` as you suggested: see 6bcaac207, let me know if that makes more sense this way. The main thing that needs to be tested is quoted strings as keys, but I thought it wouldn't hurt to add tests for all other kinds of string formatting that we may expect, just in case (also later on, we should also test other types of keys like int / float / bool if we support them).

I'm not sure I can move `test_multirun_dict_keys` to this same file though. This test specifically tests the change to `_get_value_element_as_str()` (l. 429 of types.py in the current diff of this PR), which is used in sweeps to provide the proper command line overrides. Again, the main thing that needs testing are quoted strings, but I added the other key formats as well to be safe.
Maybe to clarify the purpose of this test, I can show you how it fails if I revert the l.429 change:
```
LexerNoViableAltException: +foo={QuotedString(text='null', quote=<Quote.single: 0>):0}
```
(this is because the sweeper would not replace the `QuotedString` with its actual quoted string representation in the command line override)

Side note: I also pushed 0b532092a which removes some comments that seemed irrelevant to me (probably a copy/paste leftover from another comment you can see below them)

Reply:
Great.
There is a test in test_overrides_parser that is dedicated for that logic: `test_override_get_value_element_method`

I think testing there should be sufficient.

Reply:
> I think testing there should be sufficient.

Perfect! Done in 8084317f1

At the same time I uncovered a bug with escaped characters (that affected regular unquoted strings as well, not just dictionary keys), which is now fixed in that same commit.